frame/transaction-payment/src/lib.rs

```diff
@@ -445,6 +445,39 @@ where
 		}
 	}
 
+	/// Query runtime dispatch info of a given `Call`.
+	pub fn query_call_info<Call: GetDispatchInfo>(
```

Comment:
Why is this not `T::Call` from `system::Config`? that's exactly supposed to be the outer call. It is also to reasonable to bound `type Call` in system to be `GetDispatchInfo` rather than adding `where` here.